src/Compilers/CSharp/Portable/Symbols/PublicNullableFlowStateExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal static class PublicNullableFlowStateExtensions
+    {
+        public static NullableFlowState ToInternalFlowState(this CodeAnalysis.NullableFlowState flowState)
```

Comment:
>ToInternalFlowState [](start = 40, length = 19)

Is this necessary? When do we need to convert from the public values to internal values?

Reply:
Currently, it's being used in the SymbolDisplay visitor and the VerifyTypes helpers. I'll work on removing them in a later PR if it's possible, but it will require removing TypeWithState and TypeSymbolWithAnnotations from a few places so I'll hold off on that for now.

---
In reply to: [265784942](https://github.com/dotnet/roslyn/pull/34017#discussion_r265784942) [](ancestors = 265784942)